io_mesh_w3d/w3d/export_w3d.py

```diff
@@ -50,34 +55,47 @@ def save(self, export_settings):
 
             animation = retrieve_animation(
                 containerName, hierarchy, rig, timecoded)
-            animation.write(sknFile)
+
+            if len(animation.channels) > 0:
+                animation.write(sknFile)
 
         sknFile.close()
 
     elif export_mode == 'H':
         filename = os.path.splitext(os.path.basename(self.filepath))[0]
         sklFilePath = self.filepath.replace(
             filename, hierarchy.header.name.lower())
+        
+        if len(hierarchy.pivots) < 2:
+            self.report({'ERROR'}, "Scene does not contain any hierarchy data, aborting export!")
+            return {'CANCELLED'}
+
         sklFile = open(sklFilePath, "wb")
         hierarchy.write(sklFile)
         sklFile.close()
 
     elif export_mode == 'A':
-        aniFile = open(self.filepath, "wb")
         timecoded = False
-        compressionMode = export_settings['w3d_compression']
-
-        if compressionMode == "TC":
+        if export_settings['w3d_compression'] == "TC":
             timecoded = True
 
         animation = retrieve_animation(
             containerName, hierarchy, rig, timecoded)
+        if timecoded and not animation.time_coded_channels:
```

Comment:
This could be packed into a single if statement

Reply:
done